hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java

```diff
@@ -152,7 +174,12 @@ public long updateCommitInfosMap(
               (address, index) -> proto.getCommitIndex()))
           .filter(Objects::nonNull);
     }
-    return stream.mapToLong(Long::longValue).min().orElse(0);
+    if (level == ReplicationLevel.ALL_COMMITTED) {
+      return stream.mapToLong(Long::longValue).min().orElse(0);
+    } else {
+      // if majority committed, then find the second large index
+      return stream.sorted(Comparator.reverseOrder()).limit(majority).skip(majority - 1).findFirst().orElse(0L);
+    }
```

Comment:
We can unify the logic to return the lowest index in commitInfos, right?

Reply:
fyi In case of `MAJORITY_COMMITTED` that I observed in HDDS-10108, it is like:

| Node | Commit Index |
|------|--------------|
| A    | 8            |
| B    | 8            |
| C    | 2            |

2 nodes already goes past the commitIndex specified in watch(). With 1 node lagging behind.

Reply:
Return the lowest index is for ALL_COMMITTED watch. For majority watch, we return the second lowest index(assume factor is 3).

Reply:
In the above example, if ALL_COMMITTED, then index 2 will be returned. If MAJORITY_COMMITTED, index 8 will be returned.